crates/rpc-types-trace/src/geth/erc7562.rs

```diff
@@ -0,0 +1,331 @@
+//! Geth ERC-7562 tracer types.
+
+use alloy_primitives::{map::HashMap, Address, Bytes, B256, U256};
+use serde::{Deserialize, Serialize};
+
+use crate::geth::CallLogFrame;
+
+/// The response object for `debug_traceTransaction` with `"tracer": "erc7562Tracer"`.
+///
+/// <https://github.com/ethereum/go-ethereum/pull/31006>
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Erc7562Frame {
+    /// The accessed slots.
+    pub accessed_slots: AccessedSlots,
+    /// The contract sizes.
+    pub contract_size: HashMap<Address, ContractSize>,
+    /// The error message, if any.
+    pub error: Option<String>,
+    /// The revert reason, if any.
+    pub revert_reason: Option<String>,
+    /// The ext code access info.
+    pub ext_code_access_info: Vec<String>,
+    /// The address of the caller.
+    pub from: Address,
+    /// The amount of gas provided to the call.
+    #[serde(with = "alloy_serde::quantity")]
+    pub gas: u64,
+    /// The amount of gas used by the call.
+    #[serde(with = "alloy_serde::quantity")]
+    pub gas_used: u64,
+    /// The input of the call.
+    pub input: Bytes,
+    /// Whether the call ran out of gas.
+    pub out_of_gas: bool,
+    /// The output of the call, if any.
+    pub output: Option<Bytes>,
```

Comment:
this can be skip if Option::is_none

https://github.com/ethereum/go-ethereum/blob/355228b011ef9a85ebc0f21e7196f892038d49f0/eth/tracers/native/erc7562.go#L58-L58